packages/react-reconciler/src/ReactFiberCompleteWork.js

```diff
@@ -958,7 +961,24 @@ function completeWork(
           // Append the rendered row to the child list.
           let rendered = suspenseListState.rendering;
           if (!suspenseListState.didSuspend) {
-            suspenseListState.didSuspend = isShowingAnyFallbacks(rendered);
+            if (
+              now() > suspenseListState.tailExpiration &&
+              renderExpirationTime > Never
+            ) {
+              // We have now passed our CPU deadline and we'll just give up further
+              // attempts to render the main content and only render fallbacks.
+              // The assumption is that this is usually faster.
+              suspenseListState.didSuspend = true;
+              // Since nothing actually suspended, there will nothing to ping this
+              // to get it started back up to attempt the next item. If we can show
+              // them, then they really have the same priority as this render.
+              // So we'll pick it back up the very next render pass once we've had
+              // an opportunity to yield for paint.
+              workInProgress.expirationTime = workInProgress.childExpirationTime =
+                renderExpirationTime - 1;
```

Comment:
@bvaughn This leaves remaining work at the end similar to offscreen and deprioritization but it's not actually Never priority. Do I need to do something special to carry over interaction tracing in this case?

Reply:
Yeah see `markDidPrioritizeIdleTree`. Need to update that so it tracks the lowest remaining priority per render. And then call it here.

Reply:
Here’s the recent commit that implemented this: https://github.com/facebook/react/pull/15872/commits/603ef0e9cd2add2bd7ec7599edf9ace4045ed8ed

Reply:
Thanks Andrew!

Reply:
Yea I might need a hand with this. There is another variant of this that could happen with hydration where we want to do the inverse of this and schedule an update at a higher pri from a render and effectively restart this render.

Reply:
Looks like you got it sorted out okay 👍 but let me know if you do need a hand with the follow up or something.